sklearn/cluster/_hdbscan/_tree.pyx

```diff
@@ -282,40 +282,38 @@ cdef max_lambdas(cnp.ndarray hierarchy):
     return deaths
 
 
-cdef class TreeUnionFind (object):
+cdef class TreeUnionFind:
 
-    cdef cnp.ndarray _data_arr
-    cdef cnp.intp_t[:, ::1] _data
-    cdef cnp.ndarray is_component
+    cdef cnp.intp_t[:, ::1] data
+    cdef cnp.uint8_t[::1] is_component
 
     def __init__(self, size):
-        self._data_arr = np.zeros((size, 2), dtype=np.intp)
-        self._data_arr.T[0] = np.arange(size)
-        self._data = self._data_arr
+        cdef cnp.ndarray[cnp.intp_t, ndim=2] data_arr
+        data_arr = np.zeros((size, 2), dtype=np.intp)
+        data_arr.T[0] = np.arange(size)
+        self.data = data_arr
         self.is_component = np.ones(size, dtype=bool)
```

Comment:
A suggestion to maintain consistency between types and `dtypes`.

```suggestion
        self.is_component = np.ones(size, dtype=np.uint8)
```

What do you think?